### PR TITLE
[ROCKETMQ-74] Fix DataVersion equals defect

### PR DESCRIPTION
**Pull request by lizhanhui**

### CODE REVIEW

File: common/src/main/java/org/apache/rocketmq/common/DataVersion.java [hunk 1]
```diff
@@ -58,16 +58,28 @@ public boolean equals(final Object o) {
...
-        return counter != null ? counter.equals(that.counter) : that.counter == null;
+
+        if (null == counter && null == that.counter) {
+            return true;
+        }
+
+        if (counter != null && that.counter == null || counter == null) {
+            return false;
+        }
+
```
Comment (zhouxinyu) [inline common/src/main/java/org/apache/rocketmq/common/DataVersion.java]: I have a advise, when compare the `counter`, below code is more readable:

```java
if (counter != null && that.counter != null) {
    return counter.longValue() == that.counter.longValue();
}
return counter == that.counter;
```

**Comment by zhouxinyu** [comment]
Please @vongosling @lollipopjin help review.

**Comment by lollipopjin** [comment]
Seems ok.

**Comment by lizhanhui** [comment]
@vongosling Any opinion on this PR? If not, I would merge it to master branch.

**Comment by vongosling** [comment]
@lizhanhui since 4.1.0, we will start a new branching model, more formalized, more flexible. All PRs except those bugfix, we recommended to merge back to develop branch. This guide will publish in website sooner :-)

**Comment by djKooks** [comment]
LGTM!

**Comment by vongosling** [comment]
@djKooks You're amazing~ http://rocketmq.incubator.apache.org/docs/branching-model

**Comment by djKooks** [comment]
@vongosling Thanks for guide! I'll refer to it.

**Pull request closed without merging**